RHEL6_7/storage/dasd-format/check

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+. /usr/share/preupgrade/common.sh
+#END GENERATED SECTION
+
+ARCH="$(arch)"
+
+DASDVIEW=$(which dasdview)
+DASDVIEW=${DASDVIEW:-/sbin/dasdview}
+
+GREP=$(which grep)
+GREP=${GREP:-/bin/grep}
+
+STR_IFS=$IFS
+IFS=" $(echo -n -e '\t')"
+
+function wrap_exit
+{
+	IFS=$STR_IFS
+
+	exit $1
+}
+
+[ "$ARCH" == "s390x" ] || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $DASDVIEW || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $GREP || wrap_exit $RESULT_NOT_APPLICABLE
+test -x $CUT || wrap_exit $RESULT_NOT_APPLICABLE
+
+
+function detect_linux_disk_layout()
+{
+	log_debug "Checking Linux Disk Layout for a device $1"
+
+	local format
+	format=$($DASDVIEW -x $1 2>/dev/null | $GREP "format :" | $CUT -f 3)
+
+	test -n "$format" || {
+		log_error "dasdview parsing failed for a device $1"
+		wrap_exit $RESULT_ERROR
+	}
+
+	log_debug "Detected Direct Access Storage Device format: $format"
+
+	[[ "$format" == *"LDL"* ]]
+}
+
+function uuid_to_dev()
+{
+	$BLKID -l -o device -t $1 2>/dev/null
+} 
+
+function check_device()
+{
+	local dev
+
+	dev=$1
+
+	if [ "${dev%%=*}" = "UUID" ]; then
+		log_debug "Translating $dev to device path"
+		dev=$(uuid_to_dev $dev)
+		test -n $dev || {
+			log_error "UUID translation to device path failed."
+			wrap_exit $RESULT_ERROR 
+		}
+	fi
+
+	test -b $dev || return 0
+	detect_linux_disk_layout "$dev" && return 1
+	return 0
+}
+
+function list_devices()
+{
+	$BLKID -o device 2>/dev/null
+}
+
+function check_devices()
+{
+	local dev
+
+	log_info "Check system devices for Linux Disk Layout use."
+
+	IFS=$'\n'
+
+	for dev in $(list_devices) ; do
+		log_debug "Checking device '$dev'"
+		check_device "$dev"
+		if [ $? -ne 0 ]; then
+			log_error "A Direct Access Storage Device (DASD) was detected using Linux Disk Layout (LDL) format: $dev"
+			log_high_risk "The in-place upgrade cannot proceed further. Linux Disk Layout (LDL) format is not supported on Red Hat Enterprise Linux 7 and in-place upgrade can cause data loss."
```

Comment:
You cannot combine `log_error` and `log_high_risk`. This points more like to something that should be reported than output for user that he is not able to do upgrade. In addition, you want to inhibit the in-place upgrade so you should use `log_extreme_risk ` (the `log_high_risk` function is telling to the user, that they should do (or check) something manually before the upgrade, but can continue). Please, use `log_extreme_risk` in both cases. Maybe just something like: `A Direct Access Storage Device (DASD) was detected using Linux Disk Layout (LDL) format: $dev". Upgrade is not possible.` would be ok as more info is part of the solution.txt file. But you can keep both of them, just use `log_extreme_risk`.